context/app/static/js/components/cells/SCFindResults/hooks.ts

```diff
@@ -4,47 +4,193 @@ import { Dataset } from 'js/components/types';
 import useFindDatasetForGenes from 'js/api/scfind/useFindDatasetForGenes';
 import { useCellVariableNames } from '../MolecularDataQueryForm/hooks';
 import { useMolecularDataQueryFormTracking } from '../MolecularDataQueryForm/MolecularDataQueryFormTrackingProvider';
+import { useSelectedPathwayParticipants } from '../MolecularDataQueryForm/AutocompleteEntity/hooks';
 
 export function useSCFindCellTypeResults() {
-  const cellVariableNames = useCellVariableNames();
+  const cellTypes = useCellVariableNames();
 
   // The index of the dataset results matches the index of the cell types
   // in the original cellVariableNames array.
   const datasetsWithCellTypes = useFindDatasetsForCellTypes({
-    cellTypes: cellVariableNames,
+    cellTypes,
   });
 
   const { data = [], ...rest } = datasetsWithCellTypes;
 
+  // Get the unique cell type categories
+  // if we have "Lung.B cells" and "Lung.T cells", `lung` should be a category
+  // if we have "Lung.B cells" and "Liver.B cells", `B cells` should be a category
+  // Each individual cell type should also be a category
+  const cellTypeCategories = useMemo(() => {
+    const uniqueCellTypeCategories = new Set<string>(cellTypes);
+    cellTypes.forEach((cellType, _, arr) => {
+      const cellTypeParts = cellType.split('.');
+      // This will add `Lung` and `B cells` to the unique cell type categories
+      // if there are multiple cell types for the same organ or cell type
+      cellTypeParts.forEach((part) => {
+        if (arr.filter((ct) => ct.includes(part)).length > 1) {
+          uniqueCellTypeCategories.add(part);
+        }
+      });
+    });
+    return Array.from(uniqueCellTypeCategories);
+  }, [cellTypes]);
+
   // Map datasets to the cell type they contain
   const datasets = useMemo(() => {
-    return cellVariableNames.reduce(
+    const datasetsForEachCategory = cellTypes.reduce(
```

Comment:
Pulling into a util with some unit tests would be helpful for clarity here as well.